userguide/aws/how-tos/add-a-certificate.adoc

```diff
@@ -0,0 +1,157 @@
+= How To Add a Certificate
+
+== Overview
+
+The HTTP protocol, commonly used for message exchange over the internet, does not provide encryption or authentication. As a result messages exchanged with this protocol may be insecure and subject to eavesdropping, and their source may be uncertain. To avoid this problem, HTTP is frequently combined with an encryption scheme known as SSL, or more recently TLS, that can be used to keep messages safe from unauthorized access, and verify that the sender is who they claim to be. This secure protocol is called HTTPS.
+
+This document provides information about HTTPS and its setup that applies to all of the AWS Kill Bill architectures: single-tier, multi-tier, and cloud formation.
+
+HTTPS is based on the use of a verification document called a *certificate*. the certificate serves two purposes:
+
+1. It provides the keys necessary for a highly secure encryption.
+2. It provides the electronic signature of a *certificate authority (CA)* who guarantees the identity of the sender.
+
+Increasingly, modern browsers require valid certificates when accessing sites using the HTTPS protocol. It may be difficult or impossible to access these sites if a valid certificate is not found.
+
+The structural details of certificates are defined by international standards and are well beyond the scope of this discussion. For more details see the https://en.wikipedia.org/wiki/X.509[Wikipedia article]. One thing you need to understand, though, is the certificate authority. Normally this is either an organization that is intrinsically well-known and trusted, such as DigiCert or GlobalSign (called a *root* CA), or an organization in the *trust hierarchy* derived from a root. A CA in a trust hierarchy must be itself certified by another CA in a chain which eventually leads to the root CA. Your CA, in turn, must be able to certify that *you* are who you say you are!
+
+For temporary use, you can sometimes create a *self-signed* certificate, in which you are your own CA. Most browsers will frown on this, since you are simply vouching for yourself, but it may allow you to get started with security. We will not consider this option in this document.
+
+If you own a domain, your domain provider can serve as your CA. If you don't have a domain, you can purchase one for very low cost from a domain provider such as Godaddy. When you obtain a domain you need to prove your identity, so your domain provider can trust you. To demonstrate this trust you will need a CNAME.
+
+== Preliminary concepts
+
+=== CNAMEs
+
```

Comment:
For the information in this section, I would remove anything the reader should already know about and link out to any information that is generally available on the web, such as what a CNAME is. Any information directly pertinent to the steps below can be moved from here and put at the beginning of the section, sort of like a mini-overview. I know we're not supposed to provide excessive explanations in Tutorials, but a few sentences are sometimes needed to orient the reader or explain (briefly) why the steps are necessary.

Reply:
Based on the Divio examples (more so than their descriptions) I am trying to include the minimum explanation that seems necessary in how-tos (which this is) moreso than tutorials. @FlowWriter you would know better than me what the intended audience knows. I had some trouble understanding CNAMES but maybe they won't. But should this information go in a (Divio style) explanation document?

Reply:
As mentioned above, since you already wrote this, it won't hurt to create it as a separate explanation document. But if this type of information is easily available on the internet, you could just link to an authoritative source that explains these principles. For example, does AWS have a glossary or explanation of contents? No sense in recreating the wheel if it's not necessary.